documentation/writing_tests.md

```diff
@@ -422,6 +424,77 @@ rush add --dev -p @azure/test-utils-recorder
+
+For our Egineering Systems to pick up our tests appropriately, our packages must be configured according to their guidelines, and our tests must be thought to address their needs. In this section we will go through some of these concepts, and provide links that expand them in detail. We will be covering:
+
+- [Engineering goals](#engineering-goals).
+- [CI configuration files](#ci-configuration-files).
+- [Activating the pipeline](#activating-the-pipeline).
+- [Delivering live tests to our users](#delivering-live-tests-to-our-users).
+
+### Engineering goals
+
+Though the tests for the Azure SDK for JavaScript and TypeScript must target live resources, we should make sure they only do so when necessary. For this purpose, we must keep in mind the following guidelines:
+
+- Tests should not be flaky. Tests should pass regardless of who's executing it, when it is running, and how many times does it run.
+- While writing tests, use static resources for your tests when possible to avoid repeatedly incurring delays for setting up and tearing down resources. Once tests are pushed into the repository, the pipelines will create these resources for us. We'll examine how to set this up in the following sections.
+- Avoid calling to timed delays (like `setTimeout`) to assert that a change happened in the live resources. Also avoid locking the main thread until the resource responds.
+- The resources created in the tests should be unique. Running the same test in parallel, multiple times, should not break them.
+
+You can read more recomendations through the following link: [Best Practices for writing tests that target live resources](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/51/Testing-Guidelines).
+
+### CI configuration files
+
+To ensure that our tests are executed in the test [Azure DevOps pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) that have been previously configured by our team, some configuration files are necessary:
+
+- A file named `ci.yml` will be automatically generated at the common parent of the clients of a specific service (let's say, at the `keyvault/` level of `keyvault/keyvault-keys`).
+- A `test-resources.json` file, also at the common parent of the clients of a service, which will contain an [ARM template](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview) of the resources needed to run all of the tests of these clients.
+- And a `tests.yml` file located at the folder of each one of the clients, which is in charge of specifying what package to run, what environments to use to run the tests, and how to run the tests.
+
+To explore how to write these files, we recommend going through the guide: [Creating live tests](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/48/Create-a-new-Live-Test-pipeline?anchor=creating-live-tests).
+
+### Activating the pipeline
```

Comment:
Fortunately all of this is automated for Unified Pipelines. I believe the automation runs once per hour on `master`. So when you check in your `ci.yml` the pipelines will be automatically generated for you.

On the other hand, live test pipelines need to be set up manually for JS because they target individual packages.  (Other languages have live test pipelines per-service and those can be generated via automation)